File: src/main/java/com/lumiyaviewer/lumiya/render/shaders/FXAAProgram.java
package com.lumiyaviewer.lumiya.render.shaders;

import android.opengl.GLES20;

public class FXAAProgram extends ShaderProgram {
    public int noAAtextureSampler;
    public int texcoordOffset;
    public int textureSampler;
    public int uMVPMatrix;
    public int vPosition;
    public int vTexCoord;

    public FXAAProgram() {
        super(Shader.FXAAVertexShader, Shader.FXAAFragmentShader);
    }

    protected void bindVariables() {
        this.vPosition = GLES20.glGetAttribLocation(this.handle, "vPosition");
        this.vTexCoord = GLES20.glGetAttribLocation(this.handle, "vTexCoord");
        this.textureSampler = GLES20.glGetUniformLocation(this.handle, "textureSampler");
        this.noAAtextureSampler = GLES20.glGetUniformLocation(this.handle, "noAAtextureSampler");
        this.uMVPMatrix = GLES20.glGetUniformLocation(this.handle, "uMVPMatrix");
        this.texcoordOffset = GLES20.glGetUniformLocation(this.handle, "texcoordOffset");
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/render/shaders/FXAAProgram.kt
package com.lumiyaviewer.lumiya.render.shaders

import android.opengl.GLES20

/**
 * FXAA (Fast Approximate Anti-Aliasing) shader program.
 * Applies post-processing anti-aliasing to rendered scenes.
 */
class FXAAProgram : ShaderProgram(
    Shader.FXAAVertexShader,
    Shader.FXAAFragmentShader
) {
    var vPosition: Int = 0
    var vTexCoord: Int = 0
    var uMVPMatrix: Int = 0
    var textureSampler: Int = 0
    var noAAtextureSampler: Int = 0
    var texcoordOffset: Int = 0

    override fun bindVariables() {
        vPosition = GLES20.glGetAttribLocation(handle, "vPosition")
        vTexCoord = GLES20.glGetAttribLocation(handle, "vTexCoord")
        textureSampler = GLES20.glGetUniformLocation(handle, "textureSampler")
        noAAtextureSampler = GLES20.glGetUniformLocation(handle, "noAAtextureSampler")
        uMVPMatrix = GLES20.glGetUniformLocation(handle, "uMVPMatrix")
        texcoordOffset = GLES20.glGetUniformLocation(handle, "texcoordOffset")
    }
}
